shell/browser/ui/gtk_util.cc

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2019 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "shell/browser/ui/gtk_util.h"
+
+#include <gtk/gtk.h>
+
+namespace gtk_util {
+
+// Copied from L40-L55 in
+// https://cs.chromium.org/chromium/src/chrome/browser/ui/libgtkui/select_file_dialog_impl_gtk.cc
+#if GTK_CHECK_VERSION(3, 90, 0)
```

Comment:
This whole thing needs to be linux OS ifdef checked

Reply:
we may also want to put this alongside similar utils in `shell/common` i think!

Reply:
GTK is only used by Linux so we should avoid having it compiled for other platforms at all, we can add some rules in GN or renamed the file to `util_gtk`.

And GTK is only available in the main process, putting it under `shell/common` might not be a good idea.